libexec/rtld-elf/rtld.h

```diff
@@ -448,6 +448,15 @@ void dump_Elf_Rel(Obj_Entry *, const Elf_Rel *, u_long);
 void dump_Elf_Rela(Obj_Entry *, const Elf_Rela *, u_long);
 
 #ifdef __CHERI_PURE_CAPABILITY__
+#ifdef RTLD_SANDBOX
+/*
+ * This environment variable is exposed here so that tls.h can see it.
+ */
+extern bool ld_compartment_enable;
+
+#define	C18N_ENABLED	ld_compartment_enable
```

Comment:
Hmmm, if I do that, then an error will appear in `reloc.c:reloc_non_plt` because one of its arguments also has the name `obj_rtld`...

Reply:
Hmm, reading rtld, it seems to be quite careful to avoid exporting `obj_rtld`.  Maybe pass it as an argument to `c18n_init` instead of using it as a global?  Then you can leave it static in rtld.c.

Reply:
I've added an XXX to the `extern Obj_Entry obj_rtld`. It turns out that I don't need `obj_rtld` to be global after interrupt-safety is implemented. So this change is only a temporary hack.

Reply:
> Hmm, reading rtld, it seems to be quite careful to avoid exporting `obj_rtld`. Maybe pass it as an argument to `c18n_init` instead of using it as a global? Then you can leave it static in rtld.c.

I hadn't refreshed my page to see your last comment when I posted mine. Yes, passing `obj_rtld` as an argument is a better idea. Fix now pushed.